packages/salesforcedx-vscode-core/src/conflict/conflictDetectionService.ts

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  CliCommandExecutor,
+  Command,
+  CommandExecution,
+  CommandOutput,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import * as AdmZip from 'adm-zip';
+import * as os from 'os';
+import * as path from 'path';
+import * as shell from 'shelljs';
+import * as vscode from 'vscode';
+import { channelService } from '../channels';
+import { nls } from '../messages';
+import { notificationService, ProgressNotification } from '../notifications';
+import { taskViewService } from '../statuses';
+import { telemetryService } from '../telemetry';
+import { getRootWorkspacePath } from '../util';
+import {
+  CommonDirDirectoryDiffer,
+  DirectoryDiffer,
+  DirectoryDiffResults
+} from './directoryDiffer';
+
+export interface ConflictDetectionConfig {
+  usernameOrAlias: string;
+  packageDir: string;
+  manifest: string;
+}
+
+export class ConflictDetector {
+  private differ: DirectoryDiffer;
+  private static instance: ConflictDetector;
+
+  constructor(differ?: DirectoryDiffer) {
+    this.differ = differ || new CommonDirDirectoryDiffer();
+  }
+
+  public static getInstance(): ConflictDetector {
+    if (!ConflictDetector.instance) {
+      ConflictDetector.instance = new ConflictDetector(
+        new CommonDirDirectoryDiffer()
+      );
+    }
+    return ConflictDetector.instance;
+  }
+
+  public getCachePath(username: string): string {
+    return path.join(os.tmpdir(), '.sfdx', 'tools', 'conflicts', username);
```

Comment:
We need to add the `username` or something that makes this path unique or this won't work for users running two or more separate projects running conflict detection at once. Or it could also cause incorrect conflicts when running different projects before the temp directory gets properly cleaned.